crates/ruff_linter/src/rules/flake8_pyi/rules/duplicate_union_member.rs

```diff
@@ -104,6 +106,23 @@ pub(crate) fn duplicate_union_member<'a>(checker: &Checker, expr: &'a Expr) {
         return;
     }
 
+    // Do not reduce `Union[None, ... None]` to avoid introducing a `TypeError` unintentionally
+    // e.g. `isinstance(None, Union[None, None])`, if reduced to `isinstance(None, None)`, causes
+    // `TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union` to throw at
+    // in cases where we are reducing a `typing.Optional[None]`, our only unique node will be a `NoneLiteral`,
+    // and we want to flag on that pursuant to identifying that `typing.Optional[None]` is semantically equivalent to
+    // `None`, which is also covered by this lint.
+    if unique_nodes
+        .iter()
+        .all(|n| matches!(n, Expr::NoneLiteral(_)))
```

Comment:
I was going to leave a nit about the `n` variable name, but I think we could replace the whole closure with `is_none_literal_expr`:


```suggestion
        .all(Expr::is_none_literal_expr)
```

Reply:
unfortunately this iterator yields &&Expr so we can't do the nice little inlining function call stuff like that

going to change to `|expr| Expr::is_none_literal_expr(expr)`